realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -404,6 +404,192 @@ public boolean isValid() {
         return this;
     }
 
+    // In
+
+    /**
+     * In comparison. This allows you to test if objects match any value in an array of values.
+     *
+     * @param fieldName the field to compare.
+     * @param values values to compare with and it cannot be null or empty.
+     * @return the query object.
+     * @throws java.lang.IllegalArgumentException if values is empty or do not match class or field type.
+     */
+    public RealmQuery<E> in(String fieldName, String[] values) {
+        return in(fieldName, values, Case.SENSITIVE);
+    }
+
+    /**
+     * In comparison. This allows you to test if objects match any value in an array of values.
+     *
+     * @param fieldName the field to compare.
+     * @param values values to compare with and it cannot be null or empty.
+     * @param casing how casing is handled. {@link Case#INSENSITIVE} works only for the Latin-1 characters.
+     * @return the query object.
+     * @throws java.lang.IllegalArgumentException if values is empty or do not match class or field type.
+     */
+    public RealmQuery<E> in(String fieldName, String[] values, Case casing) {
+        if (values == null || values.length == 0) {
+            throw new IllegalArgumentException("Non-empty values must be provided.");
+        }
+        beginGroup().equalTo(fieldName, values[0], casing);
+        for (int i = 1; i < values.length; i++) {
+            or().equalTo(fieldName, values[i], casing);
+        }
+        return endGroup();
+    }
+
+    /**
+     * In comparison. This allows you to test if objects match any value in an array of values.
+     *
+     * @param fieldName the field to compare.
+     * @param values values to compare with and it cannot be null or empty.
+     * @return the query object.
+     * @throws java.lang.IllegalArgumentException if values is empty or do not match class or field type.
+     */
+    public RealmQuery<E> in(String fieldName, byte[] values) {
```

Comment:
I am a bit torn. Allowing `null` here is going to be a pain to support since we need to call another method in core for it, and I am hard pressed to see it ever being used e.g `in("myNumber", new Integer[] { null, 1, 2 })` doesn't seem like code you would normally write?

Going over our other operators, we seem to use the boxed type where it make sense, e.g `equalTo()` and `notEqualTo()`, but not in operators like `between()`. These cases are a bit more clear though as using `between()` on `null` is a clear semantic error.

I would lean towards just keeping the primitive versions for now to be honest, and just have an issue for adding boxed support if need be. I could probably be convinced if you feel strongly about it though?